File: ios-sdk/ThothSDK.swift

import Foundation

public protocol ThothListener {
    func onAction(action: String, data: String)
}

public class Thoth {
    public static var listener: ThothListener?

    public static func sendContext(_ context: String) {
        listener?.onAction(action: "process", data: "LLM response for: \(context)")
    }
}

File: android-sdk/ThothSDK.kt

package com.thoth

interface ThothListener {
    fun onAction(action: String, data: String)
}

object Thoth {
    var listener: ThothListener? = null

    fun sendContext(context: String) {
        listener?.onAction("process", "LLM response for: $context")
    }
}
